src/tools/genpolicy/src/registry.rs

```diff
@@ -215,41 +200,182 @@ impl Container {
     }
 }
 
+async fn get_config_layer(image_ref: String, socket_path: String) ->  Result<DockerConfigLayer, Box<dyn std::error::Error>>{
+    
+    let socket = socket_path.clone(); // todo: figure out how not to clone everything to get it working
+    let channel = Endpoint::try_from("http://[::]")
+        .unwrap()
+        .connect_with_connector(service_fn(move |_: Uri| UnixStream::connect(socket.clone())))
+        .await
+        .expect("Could not create client.");
```

Comment:
Are we using expect or unwrap anywhere else in this code? If not, it might be better to use ? instead of expect.